test/qa_buffer.cpp

```diff
@@ -219,7 +219,7 @@ const boost::ut::suite UserApiExamples = [] {
             const auto demoWriter = [](auto w) {
                 static int offset = 1;
                 std::iota(w.begin(), w.end(), offset);
-                offset += w.size();
+                offset += static_cast<int>(w.size());
```

Comment:
same as a above ... maybe change the `auto w` to `int32_t w` to be more explicit/expressive for future readers?!?

same for the other occurrences further below.